source/grammar/index.rst

```diff
@@ -1,12 +1,132 @@
+    return_signature: "->" classical_declaration
+    constant_declaration: "const" ( dependent_type_declaration | independent_type ) assignment 
+    classical_declaration: ( dependent_type_declaration | independent_type ) assignment?
+    program_block: "{" ( program_block | statement ) "}"
+    kernel_declaration: "kernel" id classical_type_and_id_list return_signature? ";"
+    expression_statement: expression ";"
+        :| "return" expression ";
+    expression: primary_expression 
+        :| expression binaryop expression
+        :| unary_operator expression 
+        :| membership_test
+        :| expression '[' expression ']'
+        :| call "(" expression_list? ")"
+        :| expression incrementor
+        :| quantum_measurement
+        :| expression_terminator
+    expression_terminator: real | nninteger | id
+        :| constant
+        :| time_terminator
+    constant: "pi" | "π" | "tau" | "𝜏" | "euler"
+    expression_list: ( expression "," )* expression 
+    unary_operator: "~" | "!"
+    call: expression
+        :| builtin_math
+        :| cast_operator
+    builtin_math: "sin" | "cos" | "tan" | "exp" | "ln" | "sqrt" | "popcount" | "lengthof"
+    cast_operator: declare_type
+    incrementor: "++" | "--
+    pragma: "#pragma {" ... "}"
+    declaration_statement: (quantum_declaration | classical_declaration | constant_declaration) ";"
+    assignment_expression: expression assignment_operator expression
+    binary_operator: "+" | "-" | "*" | "/" 
+        :| "<<" | ">>" | "rotr" | "rotl" | "&" | "|" | "^" | "&&" | "||"
+        :| ">" | "<" | ">=" | "<=" | "==" | "!="
+    assignment_operator: "=" | "+=" | "-=" | "*=" | "/=" |
+        :| "<<=" | ">>=" |  "&=" | "|=" | "^=" | "~=" | "->"
+    membership_test: id "in" set_declaration
+    index_set_declaration: "{" index_set_generators "}"
+    index_set_generators: expression_list | range
+    range: expression? ":" expression? ( ":" expression )?
+    selection_statement: "if (" expression ")" program_block ( "else" program_block )?
+    iteration_statement: "for" membership_test program_block | "while (" expression ")" program_block
+    selection_directive_statement: selection_directive ";"
+    selection_directive: "break" | "continue" | "exit"
+    quantum_gate_definition: "gate" quantum_gate_signature quantum_gate_block
+    quantum_gate_signature: id quantum_gate_args id_lis
+    quantum_gate_args: ( "(" classical_type_and_id_list? ")" )?
+    quantum_gate_block: "{" quantum_gate_call* "}"
+    quantum_type_and_id_list: ( quantum_type_and_id "," )* quantum_type_and_id
+    quantum_type_and_id: quantum_type designator? association 
+    quantum_statement: quantum_instruction ";"
+    quantum_instruction: quantum_gate_call
+        :| quantum_measurement
+    quantum_measurement: "measure" argument
+    quantum_gate_modifiers: ( "inv" | "pow" "[" nninteger "]" | "ctrl" ) "@"
+    quantum_gate_call: quantum_gate_name quantum_gate_args designator any_list ";"
```

Comment:
"any_list" not defined anywhere

Reply:
Also, I thought the designator is not allowed inside a gate?

Reply:
"any_list" not defined anywhere should be a list of something like `id designator?`

Reply:
> Also, I thought the designator is not allowed inside a gate?

This is use to specify the duration of the call